include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -538,14 +537,92 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
                 dealii::ExcMessage("Not implemented."));
   }
 
-
   // TURBULENCE
-  if(use_turbulence_model)
+  if(viscosity_model_data.use_turbulence_model)
   {
-    AssertThrow(turbulence_model != TurbulenceEddyViscosityModel::Undefined,
-                dealii::ExcMessage("parameter must be defined"));
-    AssertThrow(turbulence_model_constant > 0,
-                dealii::ExcMessage("parameter must be greater than zero"));
+    AssertThrow(viscosity_model_data.turbulence_model_data.turbulence_model !=
+                  TurbulenceEddyViscosityModel::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+    AssertThrow(viscosity_model_data.turbulence_model_data.constant > 0,
+                dealii::ExcMessage("Parameter must be greater than zero."));
+    AssertThrow(viscosity_model_data.treatment_of_nonlinear_viscosity !=
+                  TreatmentOfNonlinearViscosity::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+
+    if(temporal_discretization == TemporalDiscretization::BDFDualSplittingScheme)
```

Comment:
Is this a general limitation of the dual splitting scheme or just not implemented? A comment could help here.

Reply:
Regarding your first comment, I do not want to question whether variable viscosity is possible for dual splitting. I mean the explicit vs. implicit treatment of the variable viscosity field (linear vs. nonlinear system of equations).

Since an implicit treatment of the variable viscosity would require a Newton solver, this is currently not implemented for the dual splitting scheme. This is also what I would write in the error message:

"An implicit treatment of the variable viscosity field (rendering the viscous step of the dual splitting scheme nonlinear regarding the unknown velocity field) is currently not implemented for the dual splitting scheme."